src/Cli/Microsoft.DotNet.Cli.Utils/Muxer.cs

```diff
@@ -35,7 +35,18 @@ public string MuxerPath
         public Muxer()
         {
 #if NET6_0_OR_GREATER
-            _muxerPath = Environment.ProcessPath;
+            string processPath = Environment.ProcessPath;
+            string hostName = "dotnet" + Constants.ExeSuffix;
+
+            if (!processPath.EndsWith(hostName, StringComparison.OrdinalIgnoreCase))
+            {
+                var dotnetHostPath = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH") ?? Environment.GetEnvironmentVariable("DOTNET_ROOT");
+                if (!string.IsNullOrEmpty(dotnetHostPath))
+                {
+                    processPath = dotnetHostPath + Path.DirectorySeparatorChar + hostName;
+                }
+            }
+            _muxerPath = processPath;
 #else
             _muxerPath = Process.GetCurrentProcess().MainModule.FileName;
 #endif
```

Comment:
```suggestion
        public Muxer()
        {
            // SDK sets DOTNET_HOST_PATH as absolute path to current dotnet executable
            string processPath = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH");
            if (string.IsNullOrEmpty(processPath))
            {
                // fallback to DOTNET_ROOT which typically holds some dotnet executable
                var root = Environment.GetEnvironmentVariable("DOTNET_ROOT");
                if (!string.IsNullOrEmpty(root))
                {
                    processPath = Path.Combine(root, $"dotnet{Constants.ExeSuffix}");
                }
                else
                {
                    // this is a wildcard, the executing process may or may not be dotnet
#if NET
                    processPath = Environment.ProcessPath;
#else
                    processPath = Process.GetCurrentProcess().MainModule.FileName;
#endif
                }
            }

            _muxerPath = processPath;
```

Reply:
For this comment:

```
// this is a wildcard, the executing process may or may not be dotnet
```

I think that when running the product, the process will always be dotnet.  It is only when loading dotnet.dll to run tests that it could be a different process.  So I would update the comment to reflect that.

Reply:
@marcpopMSFT, please incorporate "DOTNET_HOST_PATH holds the absolute path to executable (unlike DOTNET_ROOT which points to directory)". Appending dotnet.exe to `DOTNET_HOST_PATH` will result in incorrect path. e.g. with daily build and -v:diag, I get `DOTNET_HOST_PATH = /Users/am11/.dotnet10/dotnet`

Reply:
hmm, I'm surprised the PR was passing checks if that was the case but from a few binlogs I found, that appears to be accurate. I'll make the change and let's see if the checks pass this time.